### DISCUSSION TITLE
Create line plots of Array data

### DISCUSSION THREAD

**Discussion by Philliec459** [vega/altair#3302] [General]
Hello,

I have data in an Excel file that is mostly single values per column heading for each of my samples. However, in this dataset I also have two columns that are arrays, and each in their own column. 

As shown in the attached image, I want to cross plot the porosity and permeability for each sample in each column using Altair, and also make line plots from the array as shown in the figure on the right for each sample. 

I have reformatted each array column to contain the array of data with comma separators in each cell for each sample, but that does not seem to work. I could explode the pandas DataFrame for my 2 array columns, but that creates a lot of repetition of data. I was thinking that maybe each array cell needs to start with a [ and end with a ], but that did not work. Do you have any suggestions? 

![k-phi_with_Pc_ts](https://github.com/altair-viz/altair/assets/56929340/7f17574b-19e2-4540-a47d-5dda31698c50)

**Comment by joelostblom** [comment]
Could you break this down with a simple example? For example, include a few rows of a sample dataframe, and if possible example code for a chart that you want to create

**Comment by Philliec459** [comment]
### Data:
The Excel file has 10 columns of data  for each sample row. However, we also have two cells per sample row under the column headings of Pc and Sw_Pc that are arrays having 22 values in each Excel cell separated by commas. I hope that you can see this below:

<img width="1026" alt="image" src="https://github.com/altair-viz/altair/assets/56929340/ba99f3eb-bc9b-4611-a90a-faaa0961f6b9">

Altair might be able to read in the Sw_Pc and Pc arrays directly if I formatted each array cell correctly. Now we have 22 values separated by commas and each cell starts with a [ and ends with a ].  We could start each cell with a ( and end each cell with a ) too if that would help. I have tried a few formats and one with no [] or () too, but nothing works so far. 

### Altair Code:
This is an example of a cross plot of the columns Porosity and Permeability in Altair. I want to select data points from this cross plot and then have the appropriate capillary pressure curves plotted in the second line plot for the select samples. 

```
#------------------------------------------------
#   Altair: Porosity vs. Permeability Cross Plot
#------------------------------------------------
# Brush for selection
brush = alt.selection(type='interval')

chart = alt.Chart(source).mark_circle(size=200).encode(
    alt.X('Porosity:Q',
        scale=alt.Scale(domain=(0, 0.40))),
    alt.Y('Permeability:Q',scale=alt.Scale(type='log', base=10,domain=(.01, 20000))),
    
    color=alt.condition( brush,'Texture:O', alt.value('lightgray'),
                        scale=alt.Scale(
                        domain=[ 1,         2,         3,       4,        5,        6       ],
                       range =['cyan', '#1e90ff',  'blue'  , 'orange', 'brown' , 'purple'  ])),
    
    tooltip=['image','Sample_ID','Porosity','Permeability','Lithology','Texture'],  # Must be a list for the image to render
    
).properties(
    width=500,
    height=500, 
    title="Porosity vs. Permeability Cross Plot Colored by Textural Code"
).add_selection(brush)
```

and this is an example of a capillary pressure line plot of Sw_Pc vs Pc. I tried to plot Sw_Pc vs. Pc, but this code would not work:

```
pc_curves = alt.Chart(source).mark_line().encode(
    x="Sw_Pc:Q",
    y="Pc:Q",
    color=alt.condition(brush, "Sample_ID:N", alt.value("lightgray"))
).transform_filter(brush)  # Filter based on brush selection\
```

I then tried to explode Sw_Pc and Pc for my plot and that would not work as shown below:

```
pc_curves = alt.Chart(source.explode(["Sw_Pc", "Pc"])).mark_line().encode(
    x="Sw_Pc:Q",
    y="Pc:Q",
    color=alt.condition(brush, "Sample_ID:N", alt.value("lightgray"))
).transform_filter(brush)  # Filter based on brush selection\
```

What did work was create a mark_circle or mark_point plot and remove the color to identify the samples that were selected. 

```
pc_curves = alt.Chart(source.explode(["Sw_Pc", "Pc"])).mark_point().encode(
    x="Sw_Pc:Q",
    y="Pc:Q",
    #color=alt.condition(brush, "Sample_ID:N", alt.value("lightgray"))
).transform_filter(brush)  # Filter based on brush selection\
```

What I want is to select data points from the cross plot and then show the capillary pressure curve  line plots for each of the selected samples in the second plot in Altair.
